src/lib.rs

```diff
@@ -1420,9 +1465,9 @@ mod tests {
     assert!(!glob_match("a/**/**/*", "a"));
     assert!(!glob_match("a/**/**/**/*", "a"));
     assert!(!glob_match("**/a", "a/"));
-    assert!(!glob_match("a/**/*", "a/"));
-    assert!(!glob_match("a/**/**/*", "a/"));
-    assert!(!glob_match("a/**/**/**/*", "a/"));
+    assert!(glob_match("a/**/*", "a/"));
```

Comment:
The modification of the code is inappropriate. Such a change leads to inconsistency with the behavior of the `glob-to-regex` library, which is utilized by `webpack` for handling glob patterns.

Reply:
Do you think `a/**/b` should match `a/b`?

Reply:
We should follow what `micromatch` does. Looks like that does match:

```
> require('micromatch').isMatch('a/b', 'a/**/b')
true
```

Reply:
I found this to be correct. @SyMind 
```js
> require("glob-to-regexp")("a/**/*", { globstar: true }).test("a/")
true
```
The inconsistency is that `[]` can match `path separation`.